### PR TITLE
divide ApplyTaskResultsToPipelineResults tests into success and error

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit divides ApplyTaskResultsToPipelineResults unit tests into success and error, two separate tests to improve the readability and easier to find bugs. Before this commit, the unit test doesn't check the case when the return error is not nil but expect nil, and some errors are not checked.

No functional changes, but fix some tests bugs.

/kind misc

Signed-off-by: Yongxuan Zhang yongxuanzhang@google.com

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/resources/apply_test.go [hunk 1]
```diff
@@ -3764,6 +3709,164 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
...
+	}
+}
+
+func TestApplyTaskResultsToPipelineResults_Error(t *testing.T) {
+	for _, tc := range []struct {
+		description     string
+		results         []v1beta1.PipelineResult
+		taskResults     map[string][]v1beta1.TaskRunResult
+		runResults      map[string][]v1beta1.CustomRunResult
+		expectedResults []v1beta1.PipelineRunResult
```
Comment (dibyom) [inline pkg/reconciler/pipelinerun/resources/apply_test.go:3814]: expectedResults is always nil right? Can we remove it and simplify the cmp.Diff below for the error tests?

**Review by dibyom — APPROVED** [review]
LGTM with one minor comment

File: pkg/reconciler/pipelinerun/resources/apply_test.go (hunk 1 above)
Comment (Yongxuanzhang) [reply pkg/reconciler/pipelinerun/resources/apply_test.go:3814]: Yeah I have thought about it, I think if we keep it we could make sure that for error cases we don't emit pipeline results.

Comment (Yongxuanzhang) [reply pkg/reconciler/pipelinerun/resources/apply_test.go:3814]: emmm looks like there is a case "mixed-success-tasks-some-returned-results" is returning results. 🤔

Comment (Yongxuanzhang) [reply pkg/reconciler/pipelinerun/resources/apply_test.go:3814]: It maybe a valid case when pipelinerun failed but still emit results: https://github.com/tektoncd/pipeline/issues/3749

Comment (dibyom) [reply pkg/reconciler/pipelinerun/resources/apply_test.go:3814]: ah, I missed that one

File: pkg/reconciler/pipelinerun/resources/apply_test.go [hunk 2]
```diff
@@ -3764,6 +3709,164 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
 			}},
 		},
 		expectedResults: nil,
+	}, {
+		description: "multiple-results-multiple-successful-tasks ",
```
Comment (pritidesai) [inline pkg/reconciler/pipelinerun/resources/apply_test.go:3713]: NIT: extra space after `tasks "`

File: pkg/reconciler/pipelinerun/resources/apply_test.go [hunk 3]
```diff
@@ -3764,6 +3709,164 @@ func TestApplyTaskResultsToPipelineResults(t *testing.T) {
...
+		taskResults: map[string][]v1beta1.TaskRunResult{
+			"normaltask": {{
+				Name:  "baz",
+				Value: *v1beta1.NewStructuredValues("rae"),
+			}},
+		},
+		skippedTasks: []v1beta1.SkippedTask{{
+			Name: "skippedTask",
+		}},
+		expectedResults: nil,
```
Comment (pritidesai) [inline pkg/reconciler/pipelinerun/resources/apply_test.go:3794]: same as @dibyom's comment ⬇️ expectedResults is nil by default

Comment (pritidesai) [reply pkg/reconciler/pipelinerun/resources/apply_test.go:3794]: we keep the definition but no need to initialize to nil here

**Comment by pritidesai** [comment]
thanks a bunch @Yongxuanzhang for organizing these tests, I left some NITs, please feel free to address them in a separate PR.

/lgtm

**Pull request merged**